pkg/server/container_create_test.go

```diff
@@ -610,6 +576,77 @@ func TestPrivilegedBindMount(t *testing.T) {
 	}
 }
 
+func TestMountPropagation(t *testing.T) {
+	sharedGetMountsFn := func() ([]*mount.Info, error) {
+		return []*mount.Info{
+			{
+				Mountpoint: "host-path",
+				Optional:   "shared:",
+			},
+		}, nil
+	}
+
+	slaveGetMountsFn := func() ([]*mount.Info, error) {
+		return []*mount.Info{
+			{
+				Mountpoint: "host-path",
+				Optional:   "master:",
+			},
+		}, nil
+	}
+
+	for desc, test := range map[string]struct {
+		criMount        *runtime.Mount
+		fakeGetMountsFn func() ([]*mount.Info, error)
+		optionsCheck    []string
+	}{
+		"HostPath should mount as 'rprivate' if propagation is MountPropagation_PROPAGATION_PRIVATE": {
+			criMount: &runtime.Mount{
+				ContainerPath: "container-path",
+				HostPath:      "host-path",
+				Propagation:   runtime.MountPropagation_PROPAGATION_PRIVATE,
+			},
+			fakeGetMountsFn: nil,
+			optionsCheck:    []string{"rbind", "rprivate"},
+		},
```

Comment:
Should add error test case, which host path propagation is not properly set, we should expect an error from `addOCIBindMounts`.